inc/util/string_utils.hpp

```diff
@@ -28,23 +28,25 @@ namespace ebi
     template<typename C>
     void string_split(std::string const & s, char const * delims, C & ret)
     {
-        C output;
-        char const* p = s.c_str();
-        char const* q = strpbrk(p+1, delims);
+        if (s.size() > 0) {
```

Comment:
Making this check the other way around would minimize the indentation (and the diff):

```
if (s.size() == 0) {
    return;
}

C output;
...
```